exchanges/binance/binance_websocket.go

```diff
@@ -503,130 +505,69 @@ func (b *Binance) UpdateLocalBuffer(wsdp *WebsocketDepthStream) (bool, error) {
 	return false, err
 }
 
-// generateSubscriptions generates the default subscription set
 func (b *Binance) generateSubscriptions() (subscription.List, error) {
-	var channels = make([]string, 0, len(b.Features.Subscriptions))
-	for i := range b.Features.Subscriptions {
-		name, err := channelName(b.Features.Subscriptions[i])
-		if err != nil {
-			return nil, err
-		}
-		channels = append(channels, name)
-	}
-	var subscriptions subscription.List
-	pairs, err := b.GetEnabledPairs(asset.Spot)
-	if err != nil {
-		return nil, err
-	}
-	for y := range pairs {
-		for z := range channels {
-			lp := pairs[y].Lower()
-			lp.Delimiter = ""
-			subscriptions = append(subscriptions, &subscription.Subscription{
-				Channel: lp.String() + "@" + channels[z],
-				Pairs:   currency.Pairs{pairs[y]},
-				Asset:   asset.Spot,
-			})
-		}
+	return b.Features.Subscriptions.ExpandTemplates(b)
+}
+
+// GetSubscriptionTemplateFuncs returns functions available for subscription channel templating
+func (b *Binance) GetSubscriptionTemplateFuncs() template.FuncMap {
+	return template.FuncMap{
+		"interval": formatChannelInterval,
+		"levels":   formatChannelLevels,
+		"fmt":      wsPairFmt.Format,
 	}
-	return subscriptions, nil
 }
 
-// channelName converts a Subscription Config into binance format channel suffix
-func channelName(s *subscription.Subscription) (string, error) {
-	name, ok := subscriptionNames[s.Channel]
-	if !ok {
-		return name, fmt.Errorf("%w: %s", stream.ErrSubscriptionNotSupported, s.Channel)
+func formatChannelLevels(s *subscription.Subscription) string {
+	if s.Levels != 0 {
+		return strconv.Itoa(s.Levels)
 	}
+	return ""
+}
 
+func formatChannelInterval(s *subscription.Subscription) string {
 	switch s.Channel {
 	case subscription.OrderbookChannel:
-		if s.Levels != 0 {
-			name += "@" + strconv.Itoa(s.Levels)
-		}
 		if s.Interval.Duration() == time.Second {
-			name += "@1000ms"
+			return "@1000ms"
 		} else {
-			name += "@" + s.Interval.Short()
+			return "@" + s.Interval.Short()
 		}
```

Comment:
Can drop this else and just `return "@" + s.Interval.Short()`

Reply:
Actually surprised the linter didn't bitch about that.

Fixed 942032c